library/src/window_impl.cxx

```diff
@@ -128,38 +121,39 @@ window_impl::window_impl(const options& options, const std::optional<Type>& type
   {
 #if defined(VTK_OPENGL_HAS_EGL)
     this->Internals->RenWin = vtkSmartPointer<vtkF3DEGLRenderWindow>::New();
-#else
-    throw engine::no_window_exception("Window type is EGL but VTK EGL support is not enabled");
 #endif
+    // No #else needed, it would be unreachable
```

Comment:
Why isn't it unreachable? `VTK_OPENGL_HAS_EGL` can be OFF

Reply:
Good idea, I think `#error` will help us in the future if we change something in `context` class.

Reply:
obv #error wont work as this is still compiled, I've added asserts